src/overlays/actors/ovl_En_Kaizoku/z_en_kaizoku.c

```diff
@@ -396,44 +421,45 @@ s32 func_80B85A00(EnKaizoku* this, PlayState* play, s16 arg2) {
     if (explosiveActor != NULL) {
         this->picto.actor.shape.rot.y = this->picto.actor.world.rot.y = this->picto.actor.yawTowardsPlayer;
 
-        if (((this->picto.actor.bgCheckFlags & BGCHECKFLAG_WALL) && (sp32 < 0x2EE0)) ||
+        if (((this->picto.actor.bgCheckFlags & BGCHECKFLAG_WALL) && (angleToWall < 0x2EE0)) ||
             (explosiveActor->id == ACTOR_EN_BOM_CHU)) {
             if ((explosiveActor->id == ACTOR_EN_BOM_CHU) &&
                 (Actor_WorldDistXYZToActor(&this->picto.actor, explosiveActor) < 80.0f) &&
                 (BINANG_ADD(this->picto.actor.shape.rot.y - explosiveActor->world.rot.y, 0x8000) < 0x4000)) {
-                if (this->action != KAIZOKU_ACTION_6) {
-                    func_80B87C7C(this);
+                if (this->action != KAIZOKU_ACTION_JUMP) {
+                    EnKaizoku_SetupJump(this);
                 }
             } else {
-                func_80B87900(this);
+                EnKaizoku_SetupBlock(this);
             }
 
             return true;
         }
 
-        if (this->action != KAIZOKU_ACTION_7) {
-            func_80B87E28(this);
+        if (this->action != KAIZOKU_ACTION_ROLL_BACK) {
+            EnKaizoku_SetupRollBack(this);
         }
         return true;
     }
 
     if (arg2) {
```

Comment:
Hmm I wonder if we can come up with a name for this arg2. Not quite sure what yet though